packages/core/src/client.ts

```diff
@@ -3129,27 +3133,42 @@ export class MedplumClient extends TypedEventTarget<MedplumClientEventMap> {
       url = concatUrls(this.baseUrl, url);
     }
 
-    const maxRetries = 3;
+    // Previously default for maxRetries was 3, but we will interpret maxRetries literally and not count first attempt
+    // Default of 2 matches old behavior with the new semantics
+    const maxRetries = options?.maxRetries ?? 2;
     const retryDelay = 200;
-    let response: Response | undefined = undefined;
-    for (let retry = 0; retry < maxRetries; retry++) {
+
+    // We use <= since we want to retry maxRetries times and first retry is when attemptNum === 1
+    for (let attemptNum = 0; attemptNum <= maxRetries; attemptNum++) {
       try {
         if (this.options.verbose) {
           this.logRequest(url, options);
         }
-        response = (await this.fetch(url, options)) as Response;
+        const response = (await this.fetch(url, options)) as Response;
         if (this.options.verbose) {
           this.logResponse(response);
         }
-        if (response.status < 500) {
+        // Handle non-500 response and max retries exceeded
+        // We return immediately for non-500 or 500 that has exceeded max retries
+        if (response.status < 500 || attemptNum === maxRetries) {
           return response;
         }
-      } catch (err: any) {
-        this.retryCatch(retry, maxRetries, err);
+      } catch (err) {
+        // This is for the 1st retry to avoid multiple notifications
+        if ((err as Error).message === 'Failed to fetch' && attemptNum === 0) {
+          this.dispatchEvent({ type: 'offline' });
+        }
+
+        // If we got an abort error or exceeded retries, then throw immediately
+        if ((err as Error).name === 'AbortError' || attemptNum === maxRetries) {
+          throw err;
+        }
       }
+
       await sleep(retryDelay);
     }
-    return response as Response;
```

Comment:
Before, we returned the 500 response after max retries, as that could potentially have some debugging details.  That might be worth retaining?

Reply:
👍 It is still returned after max retries, see:
https://github.com/medplum/medplum/pull/4723/files#diff-858339789c798051d9c13620bffce8ce860e7ad8ba18dba1c8711eb1f7d51a55R3153

We return before the last iteration awaits so that don't sleep an extra 200ms